### PR TITLE
Resolvers

### PR DESCRIPTION
**Pull request by riteshsp2000**
### Your checklist for this pull request

🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this
repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch**
      (right side). Don't request your default branch!
- [x] Make sure you are making a pull request against the **default branch**
      (left side). Also you should start _your branch_ off _default branch_.
- [x] Check the commit's or even all commits' message styles matches our
      requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit
      test.
- [x] I have added necessary documentation (if appropriate)

### Description
Create Queries, Mutations, Resolvers and ObjectTypes/InputTypes for the following.
- Users
- Quizzes
- Questions

Please describe your pull request.

❤️Thank you!

### Post merge checklist

- [ ] Follow steps from the [guidelines for contributing](../CONTRIBUTING.md) to
      this repository.
- [ ] If you are a new contributor, ping in the thread and one of the
      maintainers will add you to all-contributors list.

### CODE REVIEW

**Review by abhibhaw — APPROVED** [review]
@riteshsp2000 good for now, but APIs need a bit of refactoring. Maybe in the future PRs

**Comment by riteshsp2000** [comment]
Agreed totally. There needs to be some better error handling and other changes, the api style needs to be a bit more consistent. Ill refactor that in the Authentication PR

File: src/schema/user/user.resolver.ts [hunk 1]
```diff
@@ -1,32 +1,166 @@
...
+
+  /**
+   * deleterUser mutation takes in a userId parameter and deletes the corresponding user.
+   */
+  @Mutation(() => User, {
+    nullable: true,
+    description:
+      'Takes the User ObjectID as parameter and deletes the corresponding User.',
+  })
+  async deleteUser(
```
[the remark this answers is not in the document]
Comment (riteshsp2000) [reply src/schema/user/user.resolver.ts:151]: we could completely block from deleting the user. I just created it as a part of the CRUD Operation.

File: src/schema/quiz/quiz.resolver.ts [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+  })
+  async deleteQuiz(
+    @Arg('quizId', () => ObjectIdScalar) quizId: ObjectID,
+  ): Promise<Quiz | null> {
+    // TODO: Use context to allow requests only with the role of admin to proceed ahead
+    try {
+      if (!quizId) {
+        throw new Error('Bad Request: Missing Parameters');
+      }
+      return await QuizModel.findByIdAndDelete(quizId);
```
[the remark this answers is not in the document]
Comment (riteshsp2000) [reply src/schema/quiz/quiz.resolver.ts:181]: Noted. Ill make the changes accordingly in the next PR.

**Pull request merged**